### PR TITLE
Add migration guide

### LINKED ISSUE #181: Migration Log from `v1` to `v2`
We need a migration log to show the differences between v1 and v2, and how users can migrate from `v1` to `v2`.

### PR DESCRIPTION
**Pull request by Wwwsylvia**
Resolves #181

### CODE REVIEW

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+## Migrating from `v1` to `v2`
+
+The import path of `v2` is:
+```
+"oras.land/oras-go/v2"
+```
+
+Basically, you would run:
+
+```
```
Comment (shizhMSFT) [inline MIGRATION_GUIDE.md:21]: You may want ` ```sh `

File: MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+The migration can be done in an iterative fashion, as `v1` and `v2` can be imported and used at the same time.
+
+## Major Changes in `v2`
+
+- Moves `content.FileStore` to [file.Store](https://pkg.go.dev/oras.land/oras-go/v2/content/file#Store)
+- Moves `content.OCIStore` to [oci.Store](https://pkg.go.dev/oras.land/oras-go/v2/content/oci#Store)
+- Moves `content.MemoryStore` to [memory.Store](https://pkg.go.dev/oras.land/oras-go/v2/content/memory#Store)
+- Supports [Copy](https://pkg.go.dev/oras.land/oras-go/v2#Copy) with more flexible options
+- Supports [Extended Copy](https://pkg.go.dev/oras.land/oras-go/v2#ExtendedCopy) with options (experimental)
+- Supports [remote registry](https://pkg.go.dev/oras.land/oras-go/v2/registry/remote) APIs
```
Comment (shizhMSFT) [inline MIGRATION_GUIDE.md:36]: Provides SDK to access OCI-compliant and Docker-compliant registries?

File: MIGRATION_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Migration Guide
+
+In version `v2`, ORAS Go library has been completely rewritten with:
+
+- More unified interfaces
+- Fewer dependencies
+- Higher test coverage
+- Better documentation
```
Comment (shizhMSFT) [inline MIGRATION_GUIDE.md:8]: You may call out oras is now a registry client / SDK.

File: MIGRATION_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# Migration Guide
+
+In version `v2`, ORAS Go library has been completely rewritten with:
```
[the remark this answers is not in the document]
Comment (Wwwsylvia) [reply MIGRATION_GUIDE.md:3]: nice one

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**